packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -110,11 +110,26 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
     }
 
     // TODO: handle CAIP-10 addresses in proof generation of 3id-blockchain-utils
-    const account = validProof.account || validProof.address
-    let [address, chainId] = account.split('@') // eslint-disable-line prefer-const
+    const account: string = validProof.account || validProof.address
 
-    const addressCaip10 = [address, chainId].join('@')
-    if (addressCaip10.toLowerCase() !== state.metadata.controllers[0].toLowerCase()) {
+    let legacyAccountCaip10 = account
+
+    if (!account.includes('@')) {
+      const _accountSplit = account.split(':')
+      const address = _accountSplit.pop()
+      const chainId = _accountSplit.join(':')
+      legacyAccountCaip10 = `${address}@${chainId}`
+    }
+
+    let legacyControllerCaip10 = state.metadata.controllers[0]
+    if (!state.metadata.controllers[0].includes('@')) {
+      const _controllerAccountSplit = state.metadata.controllers[0].split(':')
+      const address = _controllerAccountSplit.pop()
+      const chainId = _controllerAccountSplit.join(':')
+      legacyControllerCaip10 = `${address}@${chainId}`
+    }
```

Comment:
I think we can safely assume that the controllers will always be of the legacy format.